test/test_jit_cuda_fuser.py

```diff
@@ -1985,6 +1985,135 @@ def test1(x: torch.Tensor, y: torch.Tensor):
             self.assertEqual(x.grad.dtype, x.dtype)
             self.assertEqual(y.grad.dtype, y.dtype)
 
+    @unittest.skipIf(not RUN_CUDA, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    def test_autocast_1(self):
+        def t(x: torch.Tensor, y: torch.Tensor):
+            o = x * 2.0
+            o = torch.softmax(o, dim=-1)
+            o = o * 3.0
+            o = torch.matmul(o, y)
+            return o
+
+        x = torch.randn(8, 4, dtype=torch.half, device='cuda', requires_grad=True)
+        y = torch.randn(4, 4, dtype=torch.float, device='cuda', requires_grad=True)
+        grad = torch.randn(8, 4, dtype=torch.half, device='cuda', requires_grad=False)
+        t_jit = torch.jit.script(t)
+
+        for i in range(3):
+            with torch.cuda.amp.autocast() :
+                jit_o = t_jit(x, y)
+                if i == 2 :
+                    fwd_graph = t_jit.graph_for(x, y)
+            jit_o.backward(grad)
+
+        self.assertGraphContainsExactly(fwd_graph, FUSION_GUARD, 1, consider_subgraphs=True)
+        # TODO: Can't find backward graph
```

Comment:
Is this still not working?

Reply:
fixed. Note to Kevin that retrieving graph would need your code to be under the same autocast context manager.